src/dos/dos_code_page.cpp

```diff
@@ -0,0 +1,2223 @@
+	                           const std::string& struct_name,
+	                           const uint32_t size) override;
+	void WarnOffsetPointingBackwards(const uint32_t struct_offset,
+	                                 const std::string& struct_name);
+	void WarnStructNotDirectlyAfterPrevious(const uint32_t struct_offset,
+	                                        const std::string& struct_name,
+	                                        const std::string& predecessor_name);
+
+	// Warnings, device independend
+
+	void WarnInvalidFileHeader(const uint32_t struct_offset,
+	                           const std::string& struct_name);
+	void WarnNoCodePages(const uint32_t struct_offset,
+	                     const std::string& struct_name);
+	void WarnInvalidCodePage(const uint32_t struct_offset,
+	                         const std::string& struct_name,
+	                         const uint16_t code_page);
+	void WarnInvalidDeviceName(const uint32_t struct_offset,
+	                           const std::string& struct_name);
+	void WarnUnknownDeviceType(const uint32_t struct_offset,
+	                           const std::string& struct_name);
+	void WarnNonEgaInWinNtFile(const uint32_t struct_offset,
+	                           const std::string& struct_name);
+	void WarnDuplicatedFont(const uint32_t struct_offset,
+	                        const std::string& struct_name,
+	                        const std::string& device_name,
+	                        const uint32_t code_page);
+	void WarnMultipleDevicesInSameFile();
+	void WarnScreenPrinterInSameFile();
+
+	// Warnings, screen font related
+
+	void WarnScreenFontDataSize(const uint32_t font_offset,
+	                            const uint32_t expected, const uint32_t real);
+	void WarnFontSizeDuplicateOffsets(const uint32_t struct_offset,
+	                                  const std::string& struct_name,
+	                                  const uint8_t height);
+	void WarnUnknownScreenFontFormat(const uint32_t struct_offset,
+	                                 const std::string& struct_name,
+	                                 const uint16_t version);
+	void WarnScreenFontWithAspect(const uint32_t struct_offset,
+	                              const std::string& struct_name);
+	void WarnScreenFontWidthInvalid(const uint32_t offset,
+	                                const std::string& struct_name,
+	                                const uint8_t width);
+	void WarnScreenFontDuplicateSize(const uint32_t struct_offset,
+	                                 const std::string& struct_name,
+	                                 const uint8_t height);
+	void WarnScreenFontWithoutSizes(const uint32_t struct_offset,
+	                                const std::string& struct_name);
+	void WarnScreenFontWithoutUsefulSizes(const uint32_t font_offset);
+	void WarnScreenFontSizeWithoutCharacters(const uint32_t struct_offset,
+	                                         const std::string& struct_name);
```

Comment:
Blank lines between methods, please. This just looks like a huge block of text, hard to parse where one ends and the other starts.

Reply:
OK, done.